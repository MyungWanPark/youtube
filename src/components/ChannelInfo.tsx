import React from 'react';
import { ChannelProps } from '../types/videoType';
import { useQuery } from '@tanstack/react-query';
import { useYoutube } from './../context/YoutubeApiContext';

type Props = {
    info: ChannelProps;
};

export default function ChannelInfo({ info }: Props) {
    const { youtube } = useYoutube();
    const { isLoading, error, data } = useQuery(['channelInfo', info.id], () => youtube.channelImageURL(info.id));
    return <div>{data && <img src={data} alt={info.title} />}</div>;
}
