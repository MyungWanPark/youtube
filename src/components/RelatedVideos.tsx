import React from 'react';

export default function RelatedVideos({ id }: { id: string }) {
    return <div>{id}</div>;
}
