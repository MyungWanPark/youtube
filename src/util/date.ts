import { format, register } from 'timeago.js';
import koLocale from 'timeago.js/lib/lang/ko.js';

register('ko', koLocale);

export function formatAgo(date: string, lang: string = 'en_US') {
    return format(date, lang);
}
