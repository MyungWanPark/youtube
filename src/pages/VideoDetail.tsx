import React from 'react';
import { useLocation } from 'react-router-dom';
import ChannelInfo from '../components/ChannelInfo';
import RelatedVideos from '../components/RelatedVideos';
import { VideoItem } from '../types/videoType';

export default function VideoDetail() {
    const {
        state: { video },
    }: {
        state: { video: VideoItem };
    } = useLocation();
    const { title, description, channelTitle, channelId } = video.snippet;
    return (
        <section className="flex flex-col lg:flex-row">
            <article className="basis-2/3">
                <iframe
                    title={video.snippet.title}
                    id="player"
                    width="100%"
                    height="640"
                    src={`http://www.youtube.com/embed/${video.id}`}
                />
                <div className="p-8">
                    <h2 className="font-bold text-lg">{title}</h2>
                    <ChannelInfo info={{ id: channelId, title: channelTitle }} />
                    <pre className="whitespace-pre-wrap">{description}</pre>
                </div>
            </article>
            <article className="basis-1/3">
                <RelatedVideos id={video.id} />
            </article>
        </section>
    );
}
