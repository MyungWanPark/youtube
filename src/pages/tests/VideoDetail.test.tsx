import { withRouter } from '../../test/util';
import VideoDetail from '../VideoDetail';
import { render } from '@testing-library/react';
import { videoData } from './../../test/videoData';
import ChannelInfo from './../../components/ChannelInfo';
import RelatedVideos from '../../components/RelatedVideos';
import renderer from 'react-test-renderer';

jest.mock('../../components/ChannelInfo');
jest.mock('../../components/RelatedVideos');

describe('VideoDetail', () => {
    /* afterEach(() => {
        (ChannelInfo as jest.Mock).mockReset();
        (RelatedVideos as jest.Mock).mockReset();
    }); */

    test('renders video item', () => {
        const componet = renderer.create(
            withRouter(
                [
                    {
                        path: '/',
                        element: <VideoDetail />,
                    },
                ],
                [{ pathname: '/', state: { video: videoData } }]
            )
        );

        expect(componet.toJSON()).toMatchSnapshot();
        const { title, description, channelTitle, channelId } = videoData.snippet;
    });
});
