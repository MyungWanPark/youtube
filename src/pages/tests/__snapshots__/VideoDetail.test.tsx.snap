// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoDetail renders video item 1`] = `
<section
  className="flex flex-col lg:flex-row"
>
  <article
    className="basis-2/3"
  >
    <iframe
      height="640"
      id="player"
      src="https://www.youtube.com/embed/8zIf0XvoL9Y"
      title="Barbie | Teaser Trailer"
      width="100%"
    />
    <div
      className="p-8"
    >
      <h2
        className="font-bold text-lg"
      >
        Barbie | Teaser Trailer
      </h2>
      <pre
        className="whitespace-pre-wrap"
      >
        #BarbieTheMovie, from director Greta Gerwig, only in theatres Summer 2023
      </pre>
    </div>
  </article>
  <article
    className="basis-1/3"
  />
</section>
`;
